packages/http/src/__tests__/memory-leak-prevention.spec.ts

```diff
@@ -0,0 +1,254 @@
+                },
+              },
+            },
+            '404': {
+              description: 'Not Found (Resource not found).',
+              content: {
+                'application/json': {
+                  schema: {
+                    type: 'object',
+                    additionalProperties: false,
+                    properties: {
+                      error: {
+                        type: 'string',
+                        maxLength: 50,
+                      },
+                      error_description: {
+                        type: 'string',
+                        maxLength: 4000,
+                      },
+                      status_code: {
+                        type: 'string',
+                        maxLength: 3,
+                      },
+                    },
+                  },
+                },
+              },
+            },
+            default: {
+              description: 'Default response format.',
+              content: {
+                'application/json': {
+                  schema: {
+                    type: 'object',
+                    additionalProperties: false,
+                    properties: {
+                      error: {
+                        type: 'string',
+                        maxLength: 50,
+                      },
+                      error_description: {
+                        type: 'string',
+                        maxLength: 4000,
+                      },
+                      status_code: {
+                        type: 'string',
+                        maxLength: 3,
+                      },
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  };
+
+  function round(client: PrismHttp) {
+    return client.post('/specific/echo/tests/something?complete=true', '1', { headers: { 'x-tenant-id': 'north-eu' } });
+  }
+
+  it('10k', async () => {
+    const operations = await getHttpOperationsFromSpec(SPEC);
+
+    const logger = pino({});
+    // creates special instance of pino logger that prevent collecting or logging anything
+    // Unfortunately disabled logger didn't work
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    const noop = () => {};
+    logger.info = noop;
+    logger.success = noop;
+    logger.error = noop;
+    logger.warn = noop;
+    logger.child = () => logger;
+
+    const client = createClientFromOperations(operations, {
+      validateRequest: true,
+      validateResponse: true,
+      checkSecurity: true,
+      errors: true,
+      mock: {
+        dynamic: true,
+      },
+      logger,
+    });
+
+    round(client);
+    const baseMemoryUsage = process.memoryUsage().heapUsed;
+    for (let i = 0; i < 5000; i++) {
+      round(client);
+      if (i % 1000 === 0) {
+        global.gc();
+      }
+    }
+
+    global.gc();
+    expect(process.memoryUsage().heapUsed).toBeLessThanOrEqual(baseMemoryUsage);
```

Comment:
I tried to update the loop to 10K and this test fails
```
➜  prism git:(fix/1881-memory-leak) ✗ node --expose-gc ./node_modules/.bin/jest  packages/http/src/__tests__/memory-leak-prevention.spec.ts
 FAIL   HTTP  packages/http/src/__tests__/memory-leak-prevention.spec.ts (18.415 s)
  Checks if memory leaks when handling of requests
    ✕ 10k (16724 ms)

  ● Checks if memory leaks when handling of requests › 10k

    expect(received).toBeLessThanOrEqual(expected)

    Expected: <= 105417712
    Received:    117299920

      250 |
      251 |     global.gc();
    > 252 |     expect(process.memoryUsage().heapUsed).toBeLessThanOrEqual(baseMemoryUsage);
          |                                            ^
      253 |   });
      254 | });
      255 |

      at Object.<anonymous> (packages/http/src/__tests__/memory-leak-prevention.spec.ts:252:44)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        18.461 s, estimated 23 s
Ran all test suites matching /packages\/http\/src\/__tests__\/memory-leak-prevention.spec.ts/i.
```

Maybe we should accept some percent of memory change.

Reply:
At first glance I thought that doing somehting like "retry" would be good enough but then I thought that retrying in end would take the lowest memory usage emong the tries. Well... we're already doing 5000 of them so gathering few memory usage would be enough and it works ;)